js/search.js

```diff
@@ -0,0 +1,191 @@
+      search.helper.setPage(page);
+      helper.search();
+    }
+  });
+
+  search.on('render', function () {
+    $home.hide();
+    $results.show();
+  });
+
+  $inputClear.on('click', function () {
+    search.helper.setQuery('').search();
+    $('#pkg-search-input').val('').focus();
+  });
+
+  search.addWidget(instantsearch.widgets.searchBox({
+    container: '#pkg-search-input',
+    placeholder: 'webpack, babel-plugin, lodash…'
+  }));
+
+  search.addWidget(instantsearch.widgets.currentRefinedValues({
+    container: '#pkg-search-refinements',
+    clearAll: 'after'
+  }));
+
+  search.addWidget(instantsearch.widgets.hits({
+    container: '#pkg-search-hits',
+    hitsPerPage: 10,
+    templates: {
+      item: itemTemplate,
+      empty: emptyTemplate
+    }
+  }));
+
+  search.addWidget(instantsearch.widgets.pagination({
+    container: '#pkg-search-pagination',
+    cssClasses: {
+      root: 'pagination',
+      item: 'page-item',
+      active: 'active',
+      disabled: 'disabled',
+      link: 'page-link'
+    },
+    showFirstLast: false,
+    labels: {
+      previous: 'previous',
+      next: 'next'
+    },
+    scrollTo: true
+  }));
+
+  search.start();
+
+  function itemTemplate(hit) {
+    var pkgLink = 'https://www.npmjs.com/' + encode(hit.name);
+
+    var name = '<a class="ais-hit--name" href="' + pkgLink + '">' + highlight('name', hit) + '</a>';
+
+    var popular = hit.humanDownloadsLast30Days ? '<span class="ais-hit--popular ' + getDownloadBucket(hit.downloadsLast30Days) + ('" title="Downloads last 30 days">' + hit.humanDownloadsLast30Days + '</span>') : '';
+
+    var license = hit.license ? '<span class="ais-hit--license">' + hit.license + '</span>' : '';
+
+    var version = '<span class="ais-hit--version">v' + hit.version + '</span>';
+
+    var description = '<p class="ais-hit--description">' + (hit.description ? highlight('description', hit) : 'No description found in package.json.') + '</p>';
+
+    var owner = '<a class="ais-hit--ownerLink" href="' + hit.owner.link + '">\n      <img src="' + hit.owner.avatar + '?size=40" width="20" height="20" class="ais-hit--ownerAvatar" />\n      ' + hit.owner.name + '</a>';
+
+    var keywords = 'keywords' in hit._highlightResult && hit.keywords.length > 0 ? '<span class="ais-hit--keywords hidden-sm-down">' + formatKeywords(hit._highlightResult.keywords) + '</span>' : '';
+
+    var lastUpdate = '<span class="ais-hit--lastUpdate">' + moment(hit.modified).fromNow() + '</span>';
+
+    var githubRepo = hit.githubRepo ? '<span class="ais-hit--link-github"><a href="https://github.com/' + encode(hit.githubRepo.user) + '/' + encode(hit.githubRepo.project) + hit.githubRepo.path + '" title="Github repository of ' + hit.name + '">GitHub</a></span>' : '';
+
+    var homepage = hit.homepage ? '<span class="ais-hit--link-homepage"><a href="' + hit.homepage + '" title="Homepage of ' + hit.name + '">Homepage</a></span>' : '';
```

Comment:
We ensure data is deeply sanitized inside the index here: https://github.com/algolia/npm-search/blob/master/formatPkg.js

So that the frontend search does not suffers from escaping everything frontend wise.

Do you think it's sufficient or would you like to see everything still escaped here?